woocommerce/wc-admin
# Add merchant email notifications

Fixes #5300

This PR adds a new type of admin notification. With this type of notification, it will be possible to send notes via email. The procedure to create them is the same as the other ones.

**A new type**
Now it's possible to create notes type: `email`. That kind of note is sent to a certain role (`administrator` by default). Also, it's possible to extend this functionality by adding new email templates.

**A new status**
The status `sent` was added for the notes type `email`. As soon as the note is sent it's status is set as `sent`. The other statuses work as usual, so if the CTA is activated the note status will be `actioned` (the action triggering will be [addressed in a follow-up PR](https://github.com/woocommerce/woocommerce-admin/pull/5976), here we only send the CTA link but the click is not being tracked).


### Accessibility




### Screenshots
![screenshot-one wordpress test-2020 12 30-14_26_49](https://user-images.githubusercontent.com/1314156/103371293-d6e40f00-4aad-11eb-89ef-c91fd57883f5.png)

### Detailed test instructions:
_In order to test this functionality I created the branches `add/5300_test` and `add/5300_test_plain`, with notes type `email`._

- Checkout the branch "add/5300_test" (`git checkout add/5300_test`).
- Create a zip for testing with `npm run zip:test`.
- Create a `jurassic.ninja` instance (be sure to check the `Include WP Log Viewer plugin` option in the site settings).
- Upload the plugin and activate it.
- Run the cron (this tool can help [WP Crontrol](https://wordpress.org/plugins/wp-crontrol/)).
- You should have received an email, don't open it yet.
- Go to the JN site `debug.log` file. Verify that it's empty.
- Open the email and verify that the JN site `debug.log` file says `Email opened`.

**Plain email**
- In order to force a plain email I created the branch `add/5300_test_plain` so checkout that branch.
- Create the new .zip from this branch and upload it. 
- Run the cron and verify you received the email as expected.
- No opening validation will be triggered here.
